### ISSUE TITLE
jsonwebtoken package produce an error

### ISSUE THREAD

**Issue by avg206** [yarnpkg/yarn#834]
**Do you want to request a _feature_ or report a _bug_?** - bug

**What is the current behavior?** error when installing jsonwebtoken package

```
[2/4] 🚚  Fetching packages...
error https://registry.npmjs.org/jsonwebtoken/-/jsonwebtoken-7.0.0.tgz: invalid tar file
    at Extract.Parse._startEntry (/Users/avg206/.npm-packages/lib/node_modules/yarn/node_modules/tar/lib/parse.js:149:13)
    at Extract.Parse._process (/Users/avg206/.npm-packages/lib/node_modules/yarn/node_modules/tar/lib/parse.js:131:12)
    at BlockStream.<anonymous> (/Users/avg206/.npm-packages/lib/node_modules/yarn/node_modules/tar/lib/parse.js:47:8)
    at emitOne (events.js:96:13)
    at BlockStream.emit (events.js:188:7)
    at BlockStream._emitChunk (/Users/avg206/.npm-packages/lib/node_modules/yarn/node_modules/block-stream/block-stream.js:145:10)
    at BlockStream.flush (/Users/avg206/.npm-packages/lib/node_modules/yarn/node_modules/block-stream/block-stream.js:70:8)
    at BlockStream.end (/Users/avg206/.npm-packages/lib/node_modules/yarn/node_modules/block-stream/block-stream.js:66:8)
    at Extract.Parse.end (/Users/avg206/.npm-packages/lib/node_modules/yarn/node_modules/tar/lib/parse.js:86:23)
    at UnpackStream.onend (_stream_readable.js:512:10)
```

I have following note in `package.json`: `"jsonwebtoken": "7.0.0"`

**What is the expected behavior?**

I expect yarn to install this package

**Please mention your node.js, yarn and operating system version.**
Yarn:

```
$ yarn -v
yarn install v0.15.1
```

Node.js:

```
$ node -v
v6.3.0
```

OS: macOS 10.12

**Comment by ultrafez** [comment]
Do you get the same issue when you use `npm` to install the package?

I couldn't replicate the issue you're seeing - this is how I tested it:

```
$ mkdir yarn-test && cd yarn-test
$ yarn add jsonwebtoken@7.0.0
yarn add v0.15.1
info No lockfile found.
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 📃  Building fresh packages...
success Saved lockfile.
success Saved 32 new dependencies.
├─ base64-url@1.3.2
├─ base64url@1.0.6
├─ buffer-equal-constant-time@1.0.1
├─ camelcase-keys@1.0.0
├─ camelcase@1.2.1
├─ concat-stream@1.4.10
├─ core-util-is@1.0.2
├─ ecdsa-sig-formatter@1.0.7
├─ get-stdin@4.0.1
├─ hoek@2.16.3
├─ indent-string@1.2.2
├─ inherits@2.0.3
├─ is-finite@1.0.2
├─ isarray@0.0.1
├─ isemail@1.2.0
├─ joi@6.10.1
├─ jsonwebtoken@7.0.0
├─ jwa@1.1.3
├─ jws@3.1.3
├─ map-obj@1.0.1
├─ meow@2.0.0
├─ minimist@1.2.0
├─ moment@2.15.1
├─ ms@0.7.1
├─ number-is-nan@1.0.1
├─ object-assign@1.0.0
├─ readable-stream@1.1.14
├─ repeating@1.1.3
├─ string_decoder@0.10.31
├─ topo@1.1.0
├─ typedarray@0.0.6
└─ xtend@4.0.1
✨  Done in 2.64s.
$ yarn --version
0.15.1
```

**Comment by anoff** [comment]
@avg206 which npm registry are you using? I keep getting similar errors - however they're not 100% reproducable.
However I'm running against a sinopia cache due to firewall issues. With npm it works however. Maybe yarn has stricter rules for retrying/waiting for responses from the registry than npm.

**Comment by vivekagr** [comment]
@anoff That's probably it. On my first run of yarn, I got the `invalid tar file` error but it worked well on the next run with same `package.json`.

**Comment by avg206** [comment]
@anoff @vivekagr sorry, but I still have same error. Here are my whole `package.json`:
```
 "devDependencies": {
    "async": "2.0.1",
    "autoprefixer": "6.4.0",
    "babel-plugin-react-transform": "2.0.2",
    "babel-plugin-transform-decorators-legacy": "1.3.4",
    "babel-plugin-transform-react-constant-elements": "6.9.1",
    "babel-plugin-transform-react-inline-elements": "6.8.0",
    "babel-plugin-transform-react-remove-prop-types": "0.2.9",
    "chai": "3.5.0",
    "copy-webpack-plugin": "3.0.1",
    "eslint": "3.0.1",
    "eslint-config-airbnb": "9.0.1",
    "eslint-plugin-import": "1.10.3",
    "eslint-plugin-jsx-a11y": "2.0.1",
    "eslint-plugin-react": "5.2.2",
    "exports-loader": "0.6.3",
    "extract-text-webpack-plugin": "1.0.1",
    "html-webpack-plugin": "2.22.0",
    "imports-loader": "0.6.5",
    "mocha": "2.5.3",
    "mocha-mongoose": "1.2.0",
    "mock-fs": "3.11.0",
    "postcss-loader": "0.11.0",
    "react-addons-perf": "15.3.1",
    "react-hot-loader": "1.3.0",
    "react-transform-hmr": "1.0.4",
    "should": "9.0.2",
    "sinon": "1.17.5",
    "webpack-dev-server": "1.15.0",
    "webpack-merge": "0.14.1",
    "whatwg-fetch": "1.0.0",
    "passport": "0.3.2",
    "passport-saml": "0.15.0"
  },
  "dependencies": {
    "babel-core": "6.9.1",
    "babel-eslint": "6.1.2",
    "babel-loader": "6.2.1",
    "babel-plugin-transform-runtime": "6.4.3",
    "babel-polyfill": "6.9.1",
    "babel-preset-es2015": "6.9.0",
    "babel-preset-react": "6.3.13",
    "babel-preset-stage-0": "6.3.13",
    "babel-runtime": "6.3.19",
    "blueimp-md5": "^2.3.1",
    "body-parser": "1.15.1",
    "bootstrap": "3.3.7",
    "btoa": "^1.1.2",
    "classnames": "2.2.5",
    "clean-webpack-plugin": "0.1.10",
    "codemirror": "5.17.0",
    "cookie-parser": "1.4.3",
    "cors": "2.8.0",
    "css-loader": "0.23.1",
    "express": "4.13.4",
    "fecha": "2.2.0",
    "file-loader": "0.9.0",
    "fs-extra": "0.30.0",
    "http-auth": "2.4.4",
    "immutable": "3.8.1",
    "js-cookie": "2.1.2",
    "json-loader": "0.5.4",
    "jsonwebtoken": "7.0.0",
    "less": "2.7.1",
    "less-loader": "2.2.3",
    "millisec": "0.2.0",
    "moment": "2.14.1",
    "momentjs": "1.1.15",
    "mongoose": "4.4.20",
    "mongoose-auto-increment": "5.0.1",
    "mongoose-beautiful-unique-validation": "3.0.2",
    "mongoose-sequence": "3.0.2",
    "morgan": "1.7.0",
    "nanotimer": "^0.3.14",
    "node-fetch": "1.5.3",
    "node-sass": "3.7.0",
    "nodemailer": "2.0.0",
    "paginator": "1.0.0",
    "react": "15.3.1",
    "react-codemirror": "0.2.6",
    "react-dom": "15.3.1",
    "react-js-pagination": "1.1.16",
    "react-redux": "4.4.5",
    "react-router": "2.7.0",
    "react-router-redux": "4.0.5",
    "redux": "3.5.2",
    "redux-logger": "2.6.1",
    "redux-thunk": "2.0.1",
    "sass-loader": "3.2.0",
    "serve-favicon": "2.3.0",
    "sha1": "1.1.1",
    "sticky-session": "1.1.2",
    "style-loader": "0.13.1",
    "svg-url-loader": "1.0.3",
    "url-loader": "0.5.7",
    "webpack": "1.12.12",
    "webpack-dev-middleware": "1.5.1",
    "webpack-hot-middleware": "2.6.4",
    "ws": "1.1.1"
  }
```

Maybe there are any error with dependencies.

**Comment by anoff** [comment]
@avg206 for me the issue hasn't occurred since v0.16.1. How about you?
If it's the same for you I'd suggest @kittens can close this issue?

**Comment by busches** [comment]
@kittens what's the policy on abandoned issues? No reply in 27 days.

**Comment by avg206** [comment]
@anoff @busches @kittens sorry, I forgot to answer. Issues is resolved since v0.16.1. So we can close the issue
